src/mx/TextFileSource.java

```diff
@@ -0,0 +1,45 @@
+package mx;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class TextFileSource implements SourceInterface {
+
+    protected ArrayList<Domain> domains;
+
+    protected String filename;
+
+    public TextFileSource(String inputFilename) {
+        domains = new ArrayList<>();
+        filename = inputFilename;
+    }
+
+    @Override
+    /**
+     * @throws SourceNotFoundException
+     */
+    public ArrayList<Domain> getDomainList() throws SourceNotFoundException {
+        domains.clear();
+        try{
+            readDomainsFromFile();
+        }
+        catch(IOException ex){
+            throw new SourceNotFoundException(ex.getMessage());
+        }
+
+        return domains;
```

Comment:
Always be careful returning references to mutable objects - somebody can call `add()` or `clear()` on it and your internal state will change. Use `return Collections.immutableList(domains)`